src/cdk/a11y/aria-describer.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, Optional, Renderer2} from '@angular/core';
+import {Platform} from '@angular/cdk/platform';
+import {addAriaReferencedId, removeAriaReferencedId} from './aria-reference';
+
+/**
+ * Interface used to register message elements and keep a count of how many registrations have
+ * the same message and the reference to the message element used for the aria-describedby.
+ */
+export interface RegisteredMessage {
+  messageElement: HTMLElement;
+  hostElements: HTMLElement[];
+}
+
+/** ID used for the body container where all messages are appended. */
+export const MESSAGES_CONTAINER_ID = 'cdk-aria-describedby-messages';
+
+/** Global incremental identifier for each registered message element. */
+let nextId = 0;
+
+/** Global map of all registered message elements that have been placed into the document. */
+const registeredMessages = new Map<string, RegisteredMessage>();
+
+/** Container for all registered messages. */
+let messagesContainer: HTMLElement | null = null;
+
+/**
+ * Utility that creates visually hidden elements with a message content. Useful for elements that
+ * want to use aria-describedby to further describe themselves without adding additional visual
+ * content. Must be provided with as part of a component's injection tree so that it has access
+ * to the Renderer.
+ */
+@Injectable()
+export class AriaDescriber {
+  constructor(private _platform: Platform,
+              @Optional() private _renderer: Renderer2) {
+    if (!_renderer) {
+      throw Error('AriaDescriber must be provided through a component\'s provider decorator ' +
+          'so that it has access to the Angular Renderer');
+    }
+  }
+
+  /**
+   * Adds to the host element an aria-describedby reference to a hidden element that contains
+   * the message. If the same message has already been registered, then it will reuse the created
+   * message element.
+   */
+  addDescription(hostElement: HTMLElement, message: string) {
+    if (!this._platform.isBrowser || !message.trim()) { return; }
+
+    if (!registeredMessages.get(message)) {
+      const messageElement = this._createMessageElement(message);
```

Comment:
Is there no way to use existing DOM elements as description messages? For the input we always already have a `md-hint` or `md-error` element with the message.

Maybe you could make a directive that uses this service under the hood for existing elements to use?

Something like:
```ts
<span [cdkDescribes]="listOfElementsIDescribe">...</span>
```
Or the other way around:
```ts
<span [cdkDescribedBy]="listOfElementsThatDescribeMe">...</span>
```